app.py

```diff
@@ -3,38 +3,49 @@
 """
 
 import os
-import json
 import flask
-from flask import render_template, url_for, redirect, flash
-from flask_login import (
-    login_user,
-    login_required,
-    logout_user,
-)
-from setup import app, bcrypt
+from flask import flash
+from flask_login import current_user, login_user, login_required, logout_user
+from setup import app, bcrypt, csrf
 from database import db, BTUser, Profile
 from forms import LoginForm, RegisterForm, ProfileForm
 from fdc import search
 
 
-@app.route("/", methods=["GET", "POST"])
-@login_required
+def render(**kwargs):
+    """
+    Renders the page that handles routing on the frontend.
+    """
+
+    return flask.render_template("index.html", **kwargs)
+
+
+def redirect(route):
+    """
+    Handles redirection to a route.
+    """
+    return flask.redirect(flask.url_for(route))
+
+
+@app.route("/")
 def main():
     """
     The main landing page that is shown when the app is launced.
     """
-    return flask.render_template("index.html")
+    if current_user.is_authenticated:  # avoid "must login" message
+        return redirect("index")
+    return redirect("login")
 
 
-@app.route("/index", methods=["GET", "POST"])
+@app.route("/app")
 @login_required
 def index():
     """
-    The index page of the app.
+    The main page of the app.
     """
-    mock_data = {"your": "data here"}
-    data = json.dumps(mock_data)
-    return flask.render_template("index.html", data=data)
+    # NOTE: left this here because we may need it for favorites
+    data = {"your": "data here"}
+    return render(data=data)
```

Comment:
Good thinking leaving this here if we are able to complete this stretch feature.